### DISCUSSION TITLE
Why do yo remove password authentication?

### DISCUSSION THREAD

**Discussion by jajupmochi** [community/community#52167] [New to GitHub]
Why do yo remove password authentication completely? It is so difficult to use personal token, especially when I don't have a sudo privilege and no GCM installed... Is there a simpler way for my situation or is there a tutorial with all the steps (not asking me to go to other pages or check it myself.)

Sorry for the complaint, but it take me hours to configure this on my local compute and then on HPC... I think it is a very unsmooth migration and somewhat unnecessary for many non-commercial users.

**Comment by CodeMaster7000** [comment]
Hi @jajupmochi,

These are all measures to help make GitHub more secure. Therefore, to generate your personal token, please follow these steps:

1. Log into GitHub with your username and password
2. Navigate to your GitHub account settings
3. Scroll down and click ‘Developer settings’ in the list of links to the left
4. Click the Personal access tokens link
5. Click the ‘Generate new token’ button
6. Add a ‘Note’ to describe the tokens usage
7. Set an expiration date for the GitHub personal access token
8. Select the appropriate authentication scope
9. Click the ‘Generate token’ button
10. Copy your personal access token before you navigate away from the page, as the token’s alphanumeric value will not be displayed again. 

Hope this helps and have a nice day!

P.S. If you found this answer useful, could you please mark it as an answer? I am trying to get the Galaxy Brain bronze badge.

**Comment by airtower-luna** [comment]
> It is so difficult to use personal token, especially when I don't have a sudo privilege and no GCM installed... Is there a simpler way for my situation or is there a tutorial with all the steps (not asking me to go to other pages or check it myself.)

What exactly makes it difficult for you?

If you just don't want to enter the PAT too often, a password manager and the built-in `cache` credential helper (which keeps the credentials in RAM for a configurable time) will probably be all you need. See: https://git-scm.com/book/en/v2/Git-Tools-Credential-Storage
